framework/source/class/qx/ui/window/Window.js

```diff
@@ -632,17 +679,34 @@ qx.Class.define("qx.ui.window.Window",
         this.hide();
         this.fireEvent("close");
       }
+
+      // If automatically destroying the window upon close was requested, do
+      // so now. (Note that we explicitly re-obtain the autoDestroy property
+      // value, allowing the user's close handler to enable/disable it before
+      // here.)
+      if (this.getAutoDestroy()) {
+        this.removeAll();                                    // remove contents
+        qx.event.Registration.removeAllListeners(this);      // remove listeners
+        qx.core.Init.getApplication().getRoot().remove(this);// remove self
+        this.dispose();
```

Comment:
The fact that I couldn't find anyplace that removed the window from the root was what started me on this path.

Thanks for all of the comments, guys. I'll update this PR in the coming days.